### ISSUE TITLE
Implement zlint rule 'suppressed-errors'

### ISSUE THREAD

**Issue by rockorager** [rockorager/ziglint#14]
## Rule: suppressed-errors

Disallows suppressing or otherwise mishandling caught errors. Functions that return error unions could error during normal execution — silently discarding those errors hides bugs.

The rule flags empty `catch` blocks (`catch {}`, `catch |_| {}`) that silently swallow errors. Catches that propagate, inspect, or panic on the error are allowed.

### Example

```zig
// Bad: silently swallows the error
foo() catch {};
foo() catch |_| {};

// Good: propagate
try foo();

// Good: panic with context
foo() catch |err| @panic("unexpected error");

// Good: handle the error
foo() catch |err| {
    log.err("failed: {}", .{err});
    return err;
};
```

### Implementation notes

**Complexity: AST pattern matching with contextual exceptions**

The core check is straightforward: flag catch expressions with empty bodies. However, a key exception is needed for `defer` context — `defer writer.flush() catch {}` is the idiomatic cleanup pattern in Zig and should not be flagged.

Detecting panics inside catch blocks (`@panic`, `unreachable`) is also feasible at the AST level for the common cases.

### Reference

- zlint docs: https://donisaac.github.io/zlint/docs/rules/suppressed-errors
- Category: suspicious / warning

Suggested-By: @mattrobenolt
